tests/queryHas.js

```diff
@@ -35,4 +35,29 @@ describe('Parent selector query', function () {
         const shallowChildMatches = esquery(conditional, 'IfStatement:has(> LogicalExpression.test, > Identifier[name="x"])');
         assert.equal(1, shallowChildMatches.length);
     });
+
+    it('has(>BlockStatement)', function () {
+        const matches = esquery(conditional, 'IfStatement:has(>BlockStatement)');
+        assert.equal(3, matches.length);
+    });
+    it('has(BlockStatement>ExpressionStatement)', function () {
+        const matches = esquery(conditional, 'IfStatement:has(BlockStatement>ExpressionStatement)');
+        assert.equal(3, matches.length);
+    });
+    it('fails has(>BlockStatement>ExpressionStatement)', function () {
+        const matches = esquery(conditional, 'IfStatement:has(>BlockStatement>ExpressionStatement)');
+        assert.equal(3, matches.length);
+    });
```

Comment:
The cause seems to be that `child` selector (BlockStatement>ExpressionStatement) matches on the nested node (selector.right ExpressionStatement, instead of selector.left BlockStatement). Then exactNode checks if parent node has any ancestors
https://github.com/estools/esquery/blob/07ee329d6aaa6e468114687468e5c6f28a7b7beb/esquery.js#L205
https://github.com/estools/esquery/blob/07ee329d6aaa6e468114687468e5c6f28a7b7beb/esquery.js#L130-L133

### within IfStatement:has(>BlockStatement>ExpressionStatement)
```js
selector {
  type: 'child',
  left: { type: 'exactNode' },
  right: {
    type: 'child',
    left: { type: 'identifier', value: 'BlockStatement' },
    right: { type: 'identifier', value: 'ExpressionStatement' }
  }
}
```

left matches when visiting BlockStatement - print parents and children:
`IfStatement,[BlockStatement] ⚟ [ 'ExpressionStatement' ]`
right matches when visiting ExpressionStatement - print parents and children:
`IfStatement,BlockStatement,[ExpressionStatement] ⚟ [ 'CallExpression' ]` :scream:

right is false on BlockStatement. left is false on ExpressionStatement. no match :red_circle: 

### compare that to within IfStatement:has(>BlockStatement)
```js
selector {
  type: 'child',
  left: { type: 'exactNode' },
  right: { type: 'identifier', value: 'BlockStatement' }
}
```

left matches when visiting BlockStatement - print parents and children:
`IfStatement,[BlockStatement] ⚟ [ 'ExpressionStatement' ]`
right matches when visiting BlockStatement - print parents and children:
`IfStatement,[BlockStatement] ⚟ [ 'ExpressionStatement' ]`

left && right match on the same node :heavy_check_mark:

But even this doesn't look correct (should be `[BlockStatement] ⚟ [ 'ExpressionStatement' ]`).
`:has` injects IfStatement as an ancestor :question: when checking the insides of `:has(>BlockStatement)`. Then `child` matches when visiting BlockStatement, and it checks `left(ancestry[0], ancestry.slice(1), options)`:
- ancestry = [IfStatement]
- left = matcher for exactNode
- it checks if  IfStatement has no ancestors :question: not BlockStatement